cpp/merkletree/sparse_merkle_tree.h

```diff
@@ -171,29 +172,25 @@ class SparseMerkleTree {
   // or the leaf index is 0.
   //
   // @param path the path of the leaf whose inclusion proof to return.
-  std::vector<std::string> InclusionProof(const Path& path);
+  std::vector<std::string> InclusionProof(const Path& path) const;
 
   std::string Dump() const;
 
  private:
-  // WARNING WARNING WARNING
-  // 64 < 256 !
-  // WARNING WARNING WARNING
-  // TODO(alcutter): BIGNUM probably.
-  typedef uint64_t IndexType;
+  typedef cert_trans::BigNum IndexType;
```

Comment:
That's because nobody's arguing against a byte array; the BIGNUM was meant to be an easy wrapper for effectively that with the dynamic size thing thrown in for "free", and as is often the case "free" turns out to be startlingly expensive.